### PR TITLE
EKS: update node pool CF template to latest EKS node group template

### LINKED ISSUE #2948: Upgrade EKS node pool CF template
Download the latest node pool CF template:

```
wget https://amazon-eks.s3.us-west-2.amazonaws.com/cloudformation/2020-05-08/amazon-eks-nodegroup.yaml
```

Apply the necessary changes from our templates.

(We should stay close to the official templates for easier upgrades in the future)

This is change is required by the latest EKS AMIs according to the documentation:
https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html

### PR DESCRIPTION
**Pull request by sancyx**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes #2948 
| License         | Apache 2.0

### What's in this PR?
Update EKS CloudFormation templates to align with latest EKS node group template: https://amazon-eks.s3.us-west-2.amazonaws.com/cloudformation/2020-05-08/amazon-eks-nodegroup.yaml

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)

### CODE REVIEW

File: templates/eks/amazon-eks-nodepool-cf.yaml [hunk 1]
```diff
@@ -2,16 +2,92 @@
...
+
+  NodeAutoScalingGroupMinSize:
+      Type: Number
+      Description: Minimum size of Node Group ASG.
+      Default: 1
+
+  NodeAutoScalingGroupMaxBatchSize:
+      Type: Number
+      Description: Minimum size of Node Group ASG.
+      Default: 2
```
Comment (sagikazarmark) [inline templates/eks/amazon-eks-nodepool-cf.yaml:72]: This is a custom parameter related to the upgrade process, so it does not have to be in the middle of the autoscaling settings. (Custom params could be at the end of the list?)

File: internal/cluster/distribution/eks/defaults.go [hunk 2]
```diff
@@ -113,6 +113,15 @@ var defaultImageMap = []struct {
 			"us-west-2":      "ami-000a48e69e7695a4a", // US West (Oregon).
 		},
 	},
+	{
+		constraintForVersion("1.16"),
+		map[string]string{
+			// Kubernetes Version 1.16
+			"eu-north-1": "ami-030f9071e0d32b989", // EU (Stockholm).
+			"eu-west-3":  "ami-0170bd57fa0c260da", // EU (Paris).
+			"us-east-2":  "ami-0edc51bc2f03c9dc2", // US East (Ohio).
```
Comment (sagikazarmark) [inline internal/cluster/distribution/eks/defaults.go:122]: @pbalogh-sa can help with the full list, but ideally in a separate PR.

File: internal/cluster/distribution/eks/ekscluster/eks.go [hunk 3]
```diff
@@ -288,7 +288,7 @@ func (eks *UpdateClusterAmazonEKS) Validate() error {
 
 // isValidVersion validates the given K8S version
 func isValidVersion(version string) (bool, error) {
-	constraint, err := semver.NewConstraint(">= 1.13, < 1.16")
+	constraint, err := semver.NewConstraint("> 1.13, <= 1.16")
```
Comment (sagikazarmark) [inline internal/cluster/distribution/eks/ekscluster/eks.go:291]: Removing 1.13 should be in a separate PR IMHO.

File: templates/eks/amazon-eks-nodepool-cf.yaml [hunk 4]
```diff
@@ -274,7 +274,7 @@ Resources:
         !Ref Subnets
       Tags:
       - Key: Name
-        Value: !Sub "${ClusterName}-${NodeGroupName}"
+        Value: !Sub "${ClusterName}-${NodeGroupName}-Node"
```
Comment (sagikazarmark) [inline templates/eks/amazon-eks-nodepool-cf.yaml:277]: Is this going to be backward compatible?

File: templates/eks/amazon-eks-nodepool-cf.yaml (hunk 1 above)
Comment (sancyx) [reply templates/eks/amazon-eks-nodepool-cf.yaml:72]: In latest CF template parameter are alphabetically ordered, that's why I ordered the same way to be easily comparable. We may put custom params at the end in same order.

File: internal/cluster/distribution/eks/defaults.go (hunk 2 above)
Comment (sancyx) [reply internal/cluster/distribution/eks/defaults.go:122]: The plan was to have single pr

Comment (pbalogh-sa) [reply internal/cluster/distribution/eks/defaults.go]: The list is full now.

File: templates/eks/amazon-eks-nodepool-cf.yaml (hunk 1 above)
Comment (sancyx) [reply templates/eks/amazon-eks-nodepool-cf.yaml]: reordered params to have additional params at the end

File: templates/eks/amazon-eks-nodepool-cf.yaml (hunk 4 above)
Comment (sancyx) [reply templates/eks/amazon-eks-nodepool-cf.yaml]: It's not in terms of, that asg's made with previous CF template will have different tags, however it doesn't seem to effect anything, at least nodes are joining with this new tag as well to EKS master, regardless of k8s version.
